pkg/scheduler/framework/parallelize/parallelism.go

```diff
@@ -53,13 +54,31 @@ func chunkSizeFor(n, parallelism int) int {
 
 // Until is a wrapper around workqueue.ParallelizeUntil to use in scheduling algorithms.
 // A given operation will be a label that is recorded in the goroutine metric.
-func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece workqueue.DoWorkPieceFunc, operation string) {
+func (p Parallelizer) Until(ctx context.Context, pieces int, doWorkPiece func(piece int) error, operation string) error {
 	goroutinesMetric := metrics.Goroutines.WithLabelValues(operation)
+	goroutinesDurationMetric := metrics.GoroutinesDuration.WithLabelValues(operation)
+	goroutinesExecutionSuccessMetric := metrics.GoroutinesExecutionTotal.WithLabelValues(operation, "success")
+	goroutinesExecutionErrorMetric := metrics.GoroutinesExecutionTotal.WithLabelValues(operation, "error")
```

Comment:
Metric reporting can be expensive.

Can you run the scheduler_perf suite and make sure there is no significant degradation?

Reply:
@alculquicondor 
How can I run the scheduler_perf suite ?

Reply:
https://github.com/kubernetes/kubernetes/tree/99a8a6fe258a456d51dddd3dec16d168d94e4eb1/test/integration/scheduler_perf#how-to-run

Reply:
Thanks, but I like to clarify what I should do.
After running the command `make test-integration WHAT=./test/integration/scheduler_perf ETCD_LOGLEVEL=warn KUBE_TEST_VMODULE="''" KUBE_TEST_ARGS="-run=^$$ -benchtime=1ns -bench=BenchmarkPerfScheduling"`, how should I compare results between this PR and main branch ?

Reply:
@sanposhiho @alculquicondor 
Sure, though I am not sure which one is `SchedulingThreshold`, the result looks similar between this PR and master branch. For this example,  482.810178, 71.459260 etc are schedulingThreshold, right?

SchedulingCSIPVs
This PR
```
1.0s 1429 1429 929 482.810178
2.0s 1500 1500 1000 71.459260
```

master branch
```
1.0s 1423 1423 923 470.363004
2.0s 1500 1500 1000 76.939516
```

Reply:
Did you look at the json file that is generated after the scheduler-perf execution?

Reply:
There were several dat files, but there was no json file. Seems it was not generated because several test cases were failed both my PR and master branch.
I will take a look.

```
=== FAIL: test/integration/scheduler_perf BenchmarkPerfScheduling/SchedulingPreferredPodAntiAffinity (unknown)
=== RUN   BenchmarkPerfScheduling/SchedulingPreferredPodAntiAffinity
BenchmarkPerfScheduling/SchedulingPreferredPodAntiAffinity

=== FAIL: test/integration/scheduler_perf BenchmarkPerfScheduling/SchedulingMigratedInTreePVs (unknown)
=== RUN   BenchmarkPerfScheduling/SchedulingMigratedInTreePVs
BenchmarkPerfScheduling/SchedulingMigratedInTreePVs
```

Reply:
@sanposhiho 

I ran `make test-integration WHAT=./test/integration/scheduler_perf ETCD_LOGLEVEL=warn KUBE_TEST_VMODULE="''" KUBE_TEST_ARGS="-run=^$$ -benchtime=1ns -bench=BenchmarkPerfScheduling"` but tests were failed and a json file was not generated. This is same as main branch and my PR.

```
DONE 53 tests, 13 skipped, 40 failures in 683.294s
```

To finish this test successfully, what settings are needed ??
I ran `make quick-release` and `make` command. Also I am runinng the kubernetes cluster on my docker desktop.

Reply:
I cannot tell a sure thing without logs, but maybe the timeout thingy? You can try extend it with ` KUBE_TIMEOUT="-timeout=xxxxs"`.

Reply:
> I ran make quick-release and make command. Also I am runinng the kubernetes cluster on my docker desktop.

(Just FYI though, they aren't even needed. The only necessary stuff for scheduler-perf should be etcd running.)

Reply:
Probably you could run specific test cases only, not to encounter a timeout, e.g. SchedulingBasic:
```
make test-integration WHAT=./test/integration/scheduler_perf ETCD_LOGLEVEL=warn KUBE_TEST_VMODULE="''" KUBE_TEST_ARGS="-run=^$$ -benchtime=1ns -bench=BenchmarkPerfScheduling/SchedulingBasic" SHORT='--short=false'
```
Then, you can replace the SchedulingBasic with other test cases as well. Setting 1SHORT='--short=false'` is also valuable, as it will run the biggest workload as well.

Reply:
Another possible reason I could imagine is the [threshold](https://github.com/kubernetes/kubernetes/tree/master/test/integration/scheduler_perf#scheduling-throughput-thresholds). Like if your computer isn't powerful enough, the throughput isn't enough for some test cases and scheduler-perf marked them as failed.

Reply:
Failing on threshold still generate output json for the test case.

Reply:
@sanposhiho @macsko @alculquicondor 
Finally json files were generated and results between main branch and my PR were similar based on scheduling throughput.
Therefore I think this PR is fine.

However, though json files were generated, test cases were failed.
So if you have any concerns, could you run test cases on your environment as well ??

Reply:
If the json was generated with SchedulingThroughput entry, the failure doesn't matter.

Reply:
Sure, then it should be okay